Automattic/woocommerce-payments
# String-handling improvements to pre-release publishing workflow

When tag object had a multi-line message, e.g. pointing to a multi-line commit, workflow would fail because it didn't filter out message contents.

#### Changes proposed in this Pull Request

 * Ensure escape sequences are properly handled and only a first line of multi-line message is used to create a release name.

#### Testing instructions

 * Push a tag with multi-line message, e.g.:

```bash
git tag -a -m 'Multi-line tag\\n* Line1\\nLine 2' multiline-tag && git push origin multiline-tag
```

 * Wait for a workflow to complete and check that release name doesn't have a white space in the end. Check the [workflow logs](https://github.com/Automattic/woocommerce-payments/actions/workflows/pre-release-publish-on-tag.yaml) to ensure that line break is trimmed properly in the release name, e.g.:

<img width="1118" alt="afbeelding" src="https://user-images.githubusercontent.com/683297/133435385-2a0e438d-0648-4b45-9e0a-6804069e5a33.png">

* Delete the tag with the following command:

```bash
git tag --delete multiline-tag && git push --delete origin multiline-tag
```

 * Go to the [releases page](https://github.com/Automattic/woocommerce-payments/releases), find the draft for `multiline-tag` release, open it and click on **Discard Draft** to delete it completely.

-------------------
